app/actions/handlers.py

```diff
@@ -84,8 +85,8 @@ async def action_fetch_samples(integration, action_config: FetchSamplesConfig):
 
 
 @activity_logger()
-async def action_pull_observations(integration, action_config: PullObservationsConfig):
-    logger.info(f"Executing pull_observations action with integration {integration} and action_config {action_config}...")
+async def action_pull_devices(integration, action_config: PullDevicesConfig):
```

Comment:
Just to write down here what we talked today, my suggestion is to name the top-level action `action_pull_observations` and the sub-action `action_pull_observations_for_device` or similar.